tests/system/large/test_session.py

```diff
@@ -46,7 +50,63 @@
         # ),
     ],
 )
-def test_read_gbq_for_large_tables(session: Session, query_or_table, index_col):
+def test_read_gbq_for_large_tables(
+    session: bigframes.Session, query_or_table, index_col
+):
     """Verify read_gbq() is able to read large tables."""
     df = session.read_gbq(query_or_table, index_col=index_col)
     assert len(df.columns) != 0
+
+
+def test_close(session):
+    # we will create two tables and confirm that they are deleted
+    # when the session is closed
+
+    bqclient = session.bqclient
+
+    expiration = (
+        datetime.datetime.now(datetime.timezone.utc)
+        + bigframes.constants.DEFAULT_EXPIRATION
+    )
+    full_id_1 = bigframes.session._io.bigquery.create_temp_table(session, expiration)
+    full_id_2 = bigframes.session._io.bigquery.create_temp_table(session, expiration)
+
+    with pytest.raises(google.cloud.exceptions.Conflict):
+        bqclient.create_table(full_id_1)
+    with pytest.raises(google.cloud.exceptions.Conflict):
+        bqclient.create_table(full_id_2)
+
+    session.close()
+
+    with pytest.raises(google.cloud.exceptions.NotFound):
+        bqclient.delete_table(full_id_1)
+    with pytest.raises(google.cloud.exceptions.NotFound):
+        bqclient.delete_table(full_id_2)
+
+
+# this test is prohibitively slow on our test project
```

Comment:
I'm a bit uncomfortable with this skip. I wonder if we create a session in a location our tests don't really use if it'd be faster?

Reply:
Good idea, I tried that, and in doing so I found out the cause of the slowness. And I don't like it. for list_tables (and presumably list_datasets) on bigquery.Client, I had to pass the max_results parameter in order to get the page_size parameter to work, otherwise pages were still small despite passing page_size. :(

Anyway, test works now!